ocaml/xapi-idl/cluster/cluster_interface.ml

```diff
@@ -284,6 +284,23 @@ module LocalAPI (R : RPC) = struct
       @-> returning unit_p err
       )
 
+  module UPDATES = struct
+    let get =
+      let timeout_p = Param.mk ~name:"timeout" Types.float in
+      let result_p = Param.mk ~name:"updates" Types.string in
+      declare "UPDATES.get"
+        [
+          "Get updates from corosync-notifyd, this blocking call will return \
+           when there is an update from corosync-notifyd or it is timed out"
+        ]
+        (debug_info_p @-> timeout_p @-> returning result_p err)
```

Comment:
I suggests to include the unit of the timeout in the documentation. I also assume the timeout is a time span - not a point in time.

Reply:
This is using Lwt_unix.with_timeout, which says the following:
timeout d is a thread that remains suspended for d seconds and then fails with [Lwt_unix.Timeout](https://ocsigen.org/lwt/3.2.1/api/Lwt_unix#EXCEPTIONTimeout).

Reply:
hmm, I don't see the point of using monotonic time here. The timeout parameter represents how long we should wait for the command before we give up, and there is no comparison with other clock readings at all.

Reply:
The wait should use monotonic time to know how much it has waited :)
The other reason is that the constructors for `Mtime.Span.t` enforces unit of time, which removes ambiguity regarding these

Reply:
But `Lwt_unix.with_timeout` expects a float as its timeout parameter. You mean have this `UPDATES.get` function take in a `Mtime.Span.t` and then convert it to a float when we pass it to `Lwt_unix.with_timeout`?

Reply:
Ok, if I understand correctly, this is what you mean
```
      let timeout_p = Param.mk ~name:"timeout" Types.mtime
 ```
 where we need to add a new `mtime` into the rpclib? 
 
 Or maybe I misunderstand this:
 > Can we use an Int64 here so an Mtime.Span.t can be used and deserialized?
 Where does an Int64 come into play here?

Reply:
Mtime.span provides function to convert losslessly from int64 (representing microseconds) https://ocaml.org/p/mtime/latest/doc/Mtime/Span/index.html#val-to_uint64_ns.

These could be serialized as Int64